### PR TITLE
oauth working

### PR DESCRIPTION
**Pull request by darthbatman**

### CODE REVIEW

File: HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift [hunk 1]
```diff
@@ -114,47 +114,97 @@ extension HILoginFlowController {
...
-                        let names = [attendeeInfo.firstName, attendeeInfo.lastName].compactMap { $0 } as [String]
-                        user.name = names.joined(separator: " ")
-                        user.dietaryRestrictions = attendeeInfo.diet
-
-                    case .failure:
-                        break
-                    }
-
+    func getUserTokenFromCode(code: String) {
+        let json: [String: Any] = ["code": code]
```
Comment (rauhul) [inline HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift]: This should be an APIManager endpoint, there are very few reasons to use raw URLSessionDataTasks as it hurts maintainability.

File: HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift [hunk 2]
```diff
@@ -114,47 +114,97 @@ extension HILoginFlowController {
...
-        .launch()
+        task.resume()
+    }
+}
+
+// MARK: - Login Flow
+extension HILoginFlowController {
+    func getUserFromJSON(responseData: Data, loginMethod: HILoginMethod, token: String) -> HIUser? {
+        do {
+            let json = try JSONSerialization.jsonObject(with: responseData, options: JSONSerialization.ReadingOptions.allowFragments)
```
Comment (rauhul) [inline HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift]: You should be using Codable here, raw serialization is both more error prone and less maintainable

File: HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift [hunk 3]
```diff
@@ -114,47 +114,97 @@ extension HILoginFlowController {
...
+                identifier: email,
+                isActive: true,
+                id: id,
+                name: firstName + " " + lastName,
+                dietaryRestrictions: nil
+            )
+        } catch { return nil }
+    }
+    func populateUserData(loginMethod: HILoginMethod, token: String, sender: HIBaseViewController) {
+        var request = URLRequest(url: URL(string: "https://api.hackillinois.org/user/")!)
```
Comment (rauhul) [inline HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift]: APIManager

File: HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift [hunk 4]
```diff
@@ -190,20 +240,16 @@ extension HILoginFlowController: HILoginSelectionViewControllerDelegate {
                                       withUserInfo info: String?) {
         switch selection {
         case .github:
-            print("URL::\(HIAuthService.githubLoginURL())")
-            loginSession = SFAuthenticationSession(url: HIAuthService.githubLoginURL(), callbackURLScheme: nil) { [weak self] (url, error) in
-
+            loginSession = SFAuthenticationSession(url: URL(string: "https://api.hackillinois.org/auth/github/?redirect_uri=https://test.hackillinois.org/auth/?isiOS=1")!, callbackURLScheme: "https://test.hackillinois.org/auth/?isiOS=1") { [weak self] (url, error) in
```
Comment (rauhul) [inline HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift]: Avoid raw string literals, this information does not logically make sense to reside in the viewcontroller

File: HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift [hunk 5]
```diff
@@ -114,47 +114,97 @@ extension HILoginFlowController {
...
+            guard let data = data, error == nil else {
+                return
+            }
+            let responseJSON = try? JSONSerialization.jsonObject(with: data, options: [])
+            if let responseJSON = responseJSON as? [String: Any] {
+                if let token = responseJSON["token"] as? String {
+                    let tok = token.trimmingCharacters(in: .whitespacesAndNewlines)
                     DispatchQueue.main.async {
-                        NotificationCenter.default.post(name: .loginUser, object: nil, userInfo: ["user": user])
+                        self.populateUserData(loginMethod: .github, token: tok, sender: self.loginSelectionViewController)
```
Comment (rauhul) [inline HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift]: What was the rational here, I don’t remember this code well

File: HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift (hunk 1 above)
Comment (darthbatman) [reply HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift]: Do you mean something like this?

```
static func create(title: String, description: String) -> APIRequest<HIAPIAnnouncement.Contained> {
        var announcementDict = [String: Any]()
        announcementDict["title"]       = title
        announcementDict["description"] = description
        return APIRequest<HIAPIAnnouncement.Contained>(service: self, endpoint: "", body: announcementDict, method: .POST)
    }
```

Comment (rauhul) [reply HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift]: Yeah

File: HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift (hunk 5 above)
Comment (darthbatman) [reply HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift]: as per the current login flow, the jwt is obtained for the user and then used to get user and attendee information about the user before posting to the notification center

**Comment by rauhul** [comment]
going to close in favor of #160

**Pull request closed without merging**